local-cluster/src/cluster_tests.rs

```diff
@@ -334,6 +337,53 @@ pub fn kill_entry_and_spend_and_verify_rest(
     }
 }
 
+pub fn simulate_tower(node_keypair: &Keypair, votes: Vec<(Slot, Hash)>, tower_path: PathBuf) {
+    let tower_storage = FileTowerStorage::new(tower_path);
+    let mut tower = tower_storage.load(&node_keypair.pubkey()).unwrap();
+    for (slot, hash) in votes {
+        tower.record_bank_vote_and_update_lockouts(slot, hash);
+    }
+    let saved_tower = SavedTowerVersions::from(SavedTower::new(&tower, node_keypair).unwrap());
+    tower_storage.store(&saved_tower).unwrap();
+}
+
+pub fn check_min_slot_is_rooted(
+    min_slot: Slot,
+    contact_infos: &[ContactInfo],
+    connection_cache: &Arc<ConnectionCache>,
+    test_name: &str,
+) {
+    let mut last_print = Instant::now();
+    let loop_start = Instant::now();
+    let loop_timeout = Duration::from_secs(180);
+    for ingress_node in contact_infos.iter() {
+        let (rpc, tpu) = LegacyContactInfo::try_from(ingress_node)
+            .map(|node| get_client_facing_addr(connection_cache.protocol(), node))
+            .unwrap();
+        let client = ThinClient::new(rpc, tpu, connection_cache.clone());
+        loop {
+            let root_slot = client
+                .get_slot_with_commitment(CommitmentConfig::finalized())
```

Comment:
don't know the context of where you plan to use this yet, but is this more preferable than checking `root` from `VoteState`?

Reply:
That was the other option, but this was intended as a faster replacement for `check_for_new_roots()` which goes over the client.

Directly introspecting the tower has always been more of a hack, since `LocalCluster` was always supposed to support a generic interface for interacting with any cluster, even one that's remote. So when possible, trying to avoid it